src/pxl_scripts/px/http_trace_id/script.pxl

```diff
@@ -0,0 +1,155 @@
+        df.traceparent_id, "\\1")
+    df = df[df.trace_id != ""]
+
+    df = add_trace_id_link(df, start_time)
+
+    df = df[px.contains(df.trace_id, trace_id_filter)]
+
+    # Order columns.
+    df = df['time_', 'trace_id', 'source', 'destination', 'latency', 'major_version', 'req_path',
+            'req_method', 'req_headers', 'req_body', 'req_body_size', 'resp_status',
+            'resp_message', 'resp_headers', 'resp_body', 'resp_body_size']
+
+    return df
+
+
+def add_source_dest_columns(df):
+    ''' Add source and destination columns for the HTTP request.
+
+    HTTP requests are traced server-side (trace_role==2), unless the server is
+    outside of the cluster in which case the request is traced client-side (trace_role==1).
+
+    When trace_role==2, the HTTP request source is the remote_addr column
+    and destination is the pod column. When trace_role==1, the HTTP request
+    source is the pod column and the destination is the remote_addr column.
+
+    Input DataFrame must contain trace_role, upid, remote_addr columns.
+    '''
+    df.pod = df.ctx['pod']
+    df.namespace = df.ctx['namespace']
+
+    # If remote_addr is a pod, get its name. If not, use IP address.
+    df.ra_pod = px.pod_id_to_pod_name(px.ip_to_pod_id(df.remote_addr))
+    df.is_ra_pod = df.ra_pod != ''
+    df.ra_name = px.select(df.is_ra_pod, df.ra_pod, df.remote_addr)
+
+    df.is_server_tracing = df.trace_role == 2
+    df.is_source_pod_type = px.select(df.is_server_tracing, df.is_ra_pod, True)
+    df.is_dest_pod_type = px.select(df.is_server_tracing, True, df.is_ra_pod)
+
+    # Set source and destination based on trace_role.
+    df.source = px.select(df.is_server_tracing, df.ra_name, df.pod)
+    df.destination = px.select(df.is_server_tracing, df.pod, df.ra_name)
+
+    # Filter out messages with empty source / destination.
+    df = df[df.source != '']
+    df = df[df.destination != '']
+
+    df = df.drop(['ra_pod', 'is_ra_pod', 'ra_name', 'is_server_tracing'])
+
+    return df
+
+
+def add_trace_id_link(df, start_time: str):
+    '''Modifies the trace_id column to provide a deeplink in the UI to a filtered
+    view of just the requests related to that trace id. This function assumes that
+    any data frames that have empty trace ids have been filtered prior.
+    '''
+
+    # Until PxL has support for performing the inverse of px.parse_duration,
+    # compute 2x the original start_time manually. The goal is to avoid allowing
+    # a user to load the script with a specified trace id when the requests involved
+    # have aged out of the original start_time window. If PxL supported specifying
+    # the sort order of the table visualization (descending time), this
+    # wouldn't be a concern.
+    extended_start = 2 * px.parse_duration(start_time)
+    nanos_per_min = 60000000000
+    extended_start_time = px.itoa(px.floor(extended_start / nanos_per_min)) + 'm'
```

Comment:
I think you need a `"-"` here.

Reply:
See my testing done on the PR description. I initially tried a `-` prefix, but then the time window ended up positive (and incorrect). I can validate that again, but I removed the `-` prefix when my initial testing failed.

Reply:
Oh right, because `start_time` is negative. Makes sense.